pkg/provisioner/filestorage/filestorage.go

```diff
@@ -0,0 +1,215 @@
+		target, err = fsp.getMountTargetFromID(ctx, options.Parameters[mntTargetID])
+	}
+	if err != nil {
+		glog.Errorf("Failed to retrieve mount target: %s", err)
+		return nil, err
+	}
+
+	glog.V(4).Infof("Creating export set")
+	createExportResponse, err := fileStorageClient.CreateExport(ctx, filestorage.CreateExportRequest{
+		CreateExportDetails: filestorage.CreateExportDetails{
+			ExportSetId:  target.ExportSetId,
+			FileSystemId: response.FileSystem.Id,
+			Path:         common.String("/" + *response.FileSystem.Id),
+		},
+	})
+	if err != nil {
+		glog.Errorf("Failed to create export: %v", err)
+		return nil, err
+	}
+
+	var serverIP string
+	if len(target.PrivateIpIds) == 0 {
+		glog.Errorf("Failed to find server IDs associated with the mount target to provision a persistent volume")
+		return nil, fmt.Errorf("failed to find server IDs associated with the mount target")
+	}
+	privateIPID := target.PrivateIpIds[rand.Int()%len(target.PrivateIpIds)]
+	virtualNetworkClient := fsp.client.VirtualNetwork()
+	getPrivateIPResponse, err := virtualNetworkClient.GetPrivateIp(ctx, core.GetPrivateIpRequest{
+		PrivateIpId: &privateIPID,
+	})
+	if err != nil {
+		glog.Errorf("Failed to retrieve IP address for mount target privateIpID=%q: %v", privateIPID, err)
+		return nil, err
+	}
+	serverIP = *getPrivateIPResponse.PrivateIp.IpAddress
+
+	glog.Infof("Creating persistent volume on mount target with private IP address %s", serverIP)
+	return &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: *response.FileSystem.Id,
+			Annotations: map[string]string{
+				ociVolumeID: *response.FileSystem.Id,
+				ociExportID: *createExportResponse.Export.Id,
+			},
+			Labels: map[string]string{},
+		},
+		Spec: v1.PersistentVolumeSpec{
+			PersistentVolumeReclaimPolicy: options.PersistentVolumeReclaimPolicy,
+			AccessModes:                   options.PVC.Spec.AccessModes,
+			//FIXME: fs storage doesn't enforce quota, capacity is meaningless here.
+			Capacity: v1.ResourceList{
+				v1.ResourceName(v1.ResourceStorage): options.PVC.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)],
+			},
+			PersistentVolumeSource: v1.PersistentVolumeSource{
+				NFS: &v1.NFSVolumeSource{
+					// Randomnly select IP address associated with the mount target to use for attachment
+					Server:   serverIP,
+					Path:     "/" + *response.FileSystem.Id,
+					ReadOnly: false,
+				},
+			},
+		},
+	}, nil
+}
+
+// Delete destroys a OCI volume created by Provision
+func (fsp *filesystemProvisioner) Delete(volume *v1.PersistentVolume) error {
+	exportID, ok := volume.Annotations[ociExportID]
+	if !ok {
+		return fmt.Errorf("%q annotation not found on PV", ociExportID)
+	}
+	filesystemID, ok := volume.Annotations[ociVolumeID]
+	if !ok {
+		return fmt.Errorf("%q annotation not found on PV", ociVolumeID)
+	}
+	ctx, cancel := context.WithTimeout(fsp.client.Context(), fsp.client.Timeout())
```

Comment:
You're sharing the timeout across multiple requests here which is probably not what you want.